batch.go

```diff
@@ -8,34 +8,31 @@ import (
 	"filippo.io/edwards25519"
 )
 
-// BatchVerifier holds entries of public keys, signature and a scalar which are used for batch verification.
+// BatchVerifier accumulates batch entries with `Add`, before performing batch verification with `Verify`.
 type BatchVerifier struct {
-	entries []ks
+	entries []entry
 }
 
-// ks represents the public key, signature and scalar which the caller wants to batch verify
-type ks struct {
+// item represents a batch entry with the public key, signature and scalar which the caller wants to verify
+type entry struct {
 	pubkey    ed25519.PublicKey
 	signature []byte
 	k         *edwards25519.Scalar
 }
 
-// NewBatchVerifier creates a Verifier that entries of signatures, keys and messages
-// can be added to for verification
+// NewBatchVerifier creates an empty `BatchVerifier`.
 func NewBatchVerifier() BatchVerifier {
 	return BatchVerifier{
-		entries: []ks{},
+		entries: []entry{},
 	}
 }
 
-// Add adds a (public key, signature, message) triple to the current batch.
-func (v *BatchVerifier) Add(publicKey ed25519.PublicKey, sig, message []byte) bool {
-	if l := len(publicKey); l != ed25519.PublicKeySize {
-		return false
-	}
-
-	if len(sig) != ed25519.SignatureSize || sig[63]&224 != 0 {
-		return false
+// Add adds a (public key, message, sig) triple to the current batch.
+//
+// Panics if `sig` is not 64 bytes long.
+func (v *BatchVerifier) Add(publicKey ed25519.PublicKey, message, sig []byte) {
+	if len(sig) != ed25519.SignatureSize {
+		panic("signature has wrong length")
```

Comment:
I went a bit back and forth on whether this should panic, and I am thinking it shouldn't. It would make for pretty trivial DoS vulnerabilities, and in general I regretted every API I made panic. You could instead leave the scalar nil, and return false from Verify.